### PR TITLE
Updated Workflow Interface

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
Updated Workflow Interface:
1. Added a method for internal name of the building blocks.
2. Removed Exception.
3. Updated the name of the interface to `WorkflowStep`.

### Issues Resolved
Part of https://github.com/opensearch-project/opensearch-ai-flow-framework/issues/29

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dbwiddis — COMMENTED** [review]
This part looks good.

Either need to have an internal object with an accessor for payload (data/fields/etc.) which doesn't seem to fit on an interface, or a different class to pass around the payload.  See [comments here](https://github.com/opensearch-project/opensearch-ai-flow-framework/issues/29#issuecomment-1715953261)

File: src/main/java/org/opensearch/flowframework/workflow/WorkflowStep.java [hunk 1]
```diff
@@ -13,14 +13,19 @@
...
+public interface WorkflowStep {
 
     /**
      * Triggers the processing of the building block.
      *
      * @return CompletableFuture of the building block.
-     * @throws Exception if execution fails
      */
-    CompletableFuture<Workflow> execute() throws Exception;
+    CompletableFuture<WorkflowStep> execute();
```
Comment (dbwiddis) [inline src/main/java/org/opensearch/flowframework/workflow/WorkflowStep.java:23]: Let's make this signature
```java
CompletableFuture<WorkflowData> execute(WorkflowData data);
```
and then create a [tagging interface](https://www.tutorialspoint.com/What-is-a-marker-or-tagged-interface-in-Java) `WorkflowData` that we can add functionality to later.

Comment (owaiskazi19) [reply src/main/java/org/opensearch/flowframework/workflow/WorkflowStep.java]: My gradle check took 5 mins. Pushed the change.

**Review by dbwiddis — APPROVED** [review]

**Comment by dbwiddis** [comment]
LGTM! Let's merge this and iterate as we figure out just exactly what data looks like!

**Review by ohltyler — APPROVED** [review]

**Pull request merged**